### PR TITLE
Refactor linters

### PR DESCRIPTION
**Pull request by ericvergnaud**
## Changes
Naming of current linters is inconsistent, making the existing code very hard to read.
Also, SQL linters do not derive from a specialized base linter, leading to code duplication and suboptimal performance
This PR:
 - introduces specialized SqlLinter and SqlSequentialLinter
 - renames existing linters such that their names are self-explanatory

### Linked issues
None

### Functionality
None

### Tests
- [x] ran existing unit tests

### CODE REVIEW

File: src/databricks/labs/ucx/source_code/base.py [hunk 1]
```diff
@@ -136,6 +137,35 @@ class Linter:
...
 
+class SQLLinter(Linter):
+
+    def lint(self, code: str) -> Iterable[Advice]:
+        try:
+            expressions = parse_sql(code, read='databricks')
+            for expression in expressions:
+                if not expression:
+                    continue
+                yield from self.lint_expression(expression)
```
Comment (nfx) [inline src/databricks/labs/ucx/source_code/base.py:148]:
```suggestion
            for statement in parse_sql(code, read='databricks'):
                if not statement:
                    continue
                yield from self.lint_statement(statement)
```
it's a statement (higher level), not an expression (lower-level)

File: src/databricks/labs/ucx/source_code/base.py [hunk 2]
```diff
@@ -136,6 +137,35 @@ class Linter:
     def lint(self, code: str) -> Iterable[Advice]: ...
 
 
+class SQLLinter(Linter):
```
Comment (nfx) [inline src/databricks/labs/ucx/source_code/base.py:140]:
```suggestion
class QueryLinter(Linter):
```
is this easier to read? `LL` reads/sounds difficult

Comment (ericvergnaud) [reply src/databricks/labs/ucx/source_code/base.py:140]: How about SqlLinter ? (`CREATE TABLE xxx` is not a query)

Comment (ericvergnaud) [reply src/databricks/labs/ucx/source_code/base.py]: done

**Review by nfx — APPROVED** [review]

**Pull request merged**